components/brave_new_tab/resources/lib/app_model.ts

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { AppState, BackgroundType, defaultState } from './app_state'
+
+export type AppStateListener = (state: AppState) => void
+
+export interface AppModel {
+  getState: () => AppState
+  addListener: (callback: AppStateListener) => () => void
+  getString: (key: string) => string
+  getPluralString: (key: string, count: number) => Promise<string>
+  setBackgroundsEnabled: (enabled: boolean) => void
+  setSponsoredImagesEnabled: (enabled: boolean) => void
+  selectBackground: (type: BackgroundType, value: string) => void
+  showCustomBackgroundChooser: () => Promise<boolean>
+  addCustomBackgrounds: () => Promise<void>
+  removeCustomBackground: (background: string) => Promise<void>
+}
+
+export function defaultModel(): AppModel {
+  const state = defaultState()
+  return {
+    getState() { return state },
+    addListener() { return () => {} },
+    getString(key) { return '' },
```

Comment:
dunno if we should add this to the model, as we can go through `loadTimeData` directly for it?

Reply:
This has been moved out of `app_model` (now `new_tab_model`).